src/plugins/ui_feature/insert_pivot.ts

```diff
@@ -166,4 +168,41 @@ export class InsertPivotPlugin extends UIPlugin {
       });
     }
   }
+
+  splitPivotFormula(
+    sheetId: UID,
+    col: HeaderIndex,
+    row: HeaderIndex,
+    pivotId: UID,
+    pivotTableData: PivotTableData
+  ) {
+    this.dispatch("INSERT_PIVOT", {
+      sheetId,
+      col,
+      row,
+      pivotId,
+      table: pivotTableData,
+    });
+    const table = this.getters.getCoreTable({ sheetId, col, row });
+    if (table?.type === "dynamic") {
+      const zone = positionToZone({ col, row });
+      const { cols, rows, measures, fieldsType } = pivotTableData;
+      const pivotTable = new SpreadsheetPivotTable(cols, rows, measures, fieldsType || {});
+      const colNumber = pivotTable.getNumberOfDataColumns() + 1;
+      const rowNumber = pivotTable.columns.length + pivotTable.rows.length;
```

Comment:
Why do you add row.length + columns.length to compute the rowNumber ?

Also, nitpick, but adding +1 to colNumber to use colnumber-1 after is maybe useless ?

Reply:
> Why do you add row.length + columns.length to compute the rowNumber

Sadly, it's not crystal clear how the pivot works :/
Imagine the following pivot:
|  | Alice |  | Bob |  |  |
|---|---|---|---|---|---|
|  | TRUE | FALSE | TRUE | FALSE | Total |
|  | Expected Revenue | Expected Revenue | Expected Revenue | Expected Revenue | Expected Revenue |
| January | 10 | 20 | 30 | 40 | 100 |
| February | 1 | 2 | 3 | 4 | 10 |
| Total | 11 | 22 | 33 | 44 | 110 |

`columns` is an array of each row (also an array). So the first item of `columns` if the first row, etc.
So, `columns.length` is the height of the columns header (3 in this case)

`rows` is an array for each row of the body (so columns header not included), so 3 also in this case).

So the `rowNumber` of the table is the total of the columns header and the rows of the body, so 6.

I hope it's clearer with that, but we should improve it to make it more clear :)